packages/bot-web-ui/src/components/run-panel.jsx

```diff
@@ -165,89 +165,108 @@ const StatisticsInfoModal = ({ is_mobile, is_statistics_info_modal_open, toggleS
     );
 };
 
-class RunPanel extends React.PureComponent {
-    componentDidMount() {
-        this.props.onMount();
-    }
+const RunPanel = props => {
+    const {
+        active_index,
+        currency,
+        dialog_options,
+        is_clear_stat_disabled,
+        is_dialog_open,
+        is_drawer_open,
+        is_mobile,
+        is_statistics_info_modal_open,
+        lost_contracts,
+        number_of_runs,
+        onCancelButtonClick,
+        onClearStatClick,
+        onCloseDialog,
+        onMount,
+        onOkButtonClick,
+        onRunButtonClick,
+        onUnmount,
+        setActiveTabIndex,
+        toggleDrawer,
+        toggleStatisticsInfoModal,
+        total_payout,
+        total_profit,
+        total_stake,
+        won_contracts,
+    } = props;
 
-    componentWillUnmount() {
-        this.props.onUnmount();
-    }
-    render() {
-        const {
-            active_index,
-            dialog_options,
-            is_clear_stat_disabled,
-            is_drawer_open,
-            is_mobile,
-            is_statistics_info_modal_open,
-            onClearStatClick,
-            setActiveTabIndex,
-            toggleDrawer,
-            is_dialog_open,
-            onOkButtonClick,
-            onRunButtonClick,
-            onCancelButtonClick,
-            toggleStatisticsInfoModal,
-            onCloseDialog,
-        } = this.props;
-        const content = drawerContent({ active_index, is_drawer_open, setActiveTabIndex, ...this.props });
-        const footer = drawerFooter({ is_clear_stat_disabled, onClearStatClick });
+    React.useEffect(() => {
+        onMount();
+        return () => onUnmount();
+    }, [onMount, onUnmount]);
```

Comment:
I think you. can remove parameters because componentDidMount and componentWillUnmount no need params 
```suggestion
    }, []);
```

Reply:
Just to provide extra context, it's both okay to leave them here or remove them. The linter is nagging because it doesn't know whether `onMount` and `onUnmount` will change, so it's telling you to put them in the dependency array, because if they change, they should be executed again. But since we know that these functions don't change, you can leave them out as @mehrzad-fs is saying.

Both approaches are ok 👍